crates/db_views/src/structs.rs

```diff
@@ -694,22 +795,22 @@ pub struct PersonPostMentionView {
 /// A comment reply view.
 pub struct CommentReplyView {
   pub comment_reply: CommentReply,
+  pub recipient: Person,
   pub comment: Comment,
   pub creator: Person,
   pub post: Post,
   pub community: Community,
-  pub recipient: Person,
-  pub creator_banned_from_community: bool,
-  pub banned_from_community: bool,
-  pub creator_is_moderator: bool,
-  pub creator_is_admin: bool,
-  pub subscribed: SubscribedType,
   #[cfg_attr(feature = "full", ts(optional))]
-  /// The time when the comment was saved.
-  pub saved: Option<DateTime<Utc>>,
-  pub creator_blocked: bool,
+  pub community_actions: Option<CommunityActions>,
   #[cfg_attr(feature = "full", ts(optional))]
-  pub my_vote: Option<i16>,
+  pub comment_actions: Option<CommentActions>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub person_actions: Option<PersonActions>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub instance_actions: Option<InstanceActions>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub creator_community_actions: Option<CommunityActions>,
+  pub creator_is_admin: bool,
   pub can_mod: bool,
```

Comment:
If we return these action structs directly from the database, it means we have to keep this same data format for the whole lifetime of Lemmy 1.0. It will also require more effort from client devs to adjust their code. So it would be preferable to return the data in a similar form as before, including `CommentReplyView.subscribed` fields etc.

Reply:
> If we return these action structs directly from the database, it means we have to keep this same data format for the whole lifetime of Lemmy 1.0.

Since all the fields (except the primary keys like `person_id, community_id` for `CommunityActions`) are nullable on these actions structs, they're fairly future-proof. If anything the way we did it before was more dangerous, by adding and removing required fields. 

> It will also require more effort from client devs to adjust their code.

That's unavoidable, but it makes our job a ton easier, and removes all the complication of transforming potentially data from the actions structs and needing to remember to do them for every table. Reread over my PR notes as to why this is much better.